### ISSUE TITLE
Disallow leading zeros on issue IDs by default

### ISSUE THREAD

**Issue by RebeccaStevens** [mtfoley/pr-compliance-action#346]
## Describe the bug
By default, leading zeros are currently allowed. They shouldn't be.

## To Reproduce
Add leading zeros to issue ID `fix #003`

## Workflow File Contents
```yml
jobs:
  compliance:
    runs-on: ubuntu-latest
    steps:
      - uses: mtfoley/pr-compliance-action@main
```

## Expected behavior
Leading zeros should not be allowed.

## Additional context
Some issue templates like to use `fix #000` in them. They then expect the PR author to modify this. If they don't compliance shouldn't be marked as meet.

## Suggested change

Change the default `body-regex` from `(fix|resolv|clos)(e)*(s|d)* #\d+` to `(fix|resolv|clos)(e)*(s|d)* #[1-9]\d*`.

### Better Regex:

`(fix(es|ed)?|((resolve|close)(s|d)?)) #[1-9]\d*`

This regex also fixes the issue of thing like `fixs #1` and `closeeeeeeeeeeeeeeeeeeeed #1` being valid.

**Comment by mtfoley** [comment]
Hi @RebeccaStevens thanks for filing this!  If you're interested in submitting a PR for this feature, go for it.

That said, I'd point out a couple of things:
1) currently, the action doesn't do any sort of interrogation of the issue being referred to by a PR (although a fix for #00311 would probably change that).  For example, we don't even check if it's an open issue, currently. If in the future we do gather data about the issue(s) being linked, this will be a good thing to look out for!
2) the issue you filed made me notice that GitHub's web UI sanitizes the issue numbers with leading zeros like that, and even renders the link with summary and all. For example, I linked that issue as just `#00311` in that point above. The point being, if GitHub lets people get away with leading zeros, this action should too, in my opinion. 
3) I do like the improvements on the regex for the verbs fix/close/resolve. Good catch! As long as we are in line with the verbs described here, I'm good with it: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword. The way that verb list changes when you change the language on the page is actually kind of strange, because some (not all) of the English verbs change.

**Comment by RebeccaStevens** [comment]
Would this be a better regex for the issue number? `#\d*[1-9]\d*`?
It'll simple enforce the issue number isn't just a string of zeros.
